### PR TITLE
app/vmselect/netstorage: do not retry request when complexity limit i…

### PR DESCRIPTION
**Pull request by Haleygo**
…s already exceeded

### CODE REVIEW

File: app/vmselect/netstorage/netstorage.go [hunk 1]
```diff
@@ -2259,9 +2259,11 @@ func (sn *storageNode) execOnConnWithPossibleRetry(qt *querytracer.Tracer, funcN
 	}
 	var er *errRemote
 	var ne net.Error
-	if errors.As(err, &er) || errors.As(err, &ne) && ne.Timeout() || deadline.Exceeded() {
+	var le *limitExceededErr
+	if errors.As(err, &le) || errors.As(err, &er) || errors.As(err, &ne) && ne.Timeout() || deadline.Exceeded() {
```
Comment (hagen1778) [inline app/vmselect/netstorage/netstorage.go:2263]: I don't seem to find at which conditions this func could receive `limitExceededErr `. Looks like rpc actions could return either network errors or `errRemote` errors.

Comment (Haleygo) [reply app/vmselect/netstorage/netstorage.go:2263]: Now there could be limitExceededErr when prcessBlock()
https://github.com/VictoriaMetrics/VictoriaMetrics/blob/542f2a5cc4723d9c72be8fd44e185a55e14dff27/app/vmselect/netstorage/netstorage.go#L1686-L1690

This error will be eliminated after https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6464 merges to cluster branch.

Comment (Haleygo) [reply app/vmselect/netstorage/netstorage.go:2269]: Reopen this since https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6464 is refused.

**Comment by Haleygo** [comment]
PTAL @hagen1778 @f41gh7

**Review by hagen1778 — APPROVED** [review]

**Pull request merged**